src/main/java/dev/tycho/stonks/scheduledtasks/SubscriptionCheckTask.java

```diff
@@ -0,0 +1,112 @@
+      if (!Stonks.economy.withdrawPlayer(player, service.getCost()).transactionSuccess()) {
+        //Player does not have enough money
+        System.out.println("(failed - they didn't have enough money)");
+        if (onlinePlayer != null)
+          onlinePlayer.sendMessage(ChatColor.RED + "Failed to renew. You don't have enough money, please get more then manually resubscribe");
+      } else {
+        //Payment success
+        //Update that the subscription is paid
+        subscription.registerPaid();
+        try {
+          //Update the subscription in the database
+          DatabaseHelper.getInstance().getDatabaseManager().getSubscriptionDao().update(subscription);
+
+          //Pay and update the account
+          service.getAccount().getAccount().addBalance(service.getCost());
+
+          //Update the account in the database and add a log
+          DatabaseHelper.getInstance().getDatabaseManager().updateAccount(service.getAccount().getAccount());
+          DatabaseHelper.getInstance().getDatabaseManager().logTransaction(new Transaction(service.getAccount(),
+              player.getUniqueId(), "Subscription payment for " + service.getName(), service.getCost()));
+
+          if (onlinePlayer != null) {
+            onlinePlayer.sendMessage(ChatColor.GREEN +
+                "Success! Your subscription will be auto-renewed again in " + ChatColor.YELLOW + service.getDuration()
+                + ChatColor.GREEN + " days time. You can cancel at any time by doing /stonks subscriptions");
+          } else {
+            Bukkit.getServer().dispatchCommand(Bukkit.getServer().getConsoleSender(), "mail send " +
+                player.getName() + " you have been billed $" + service.getCost() + " for the service " + service.getName());
+          }
+          System.out.println("(success)");
+        } catch (SQLException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+  }
+  private void cancelSubscriptionIfOverdue(Subscription subscription) {
+    Service service = subscription.getService();
+    OfflinePlayer player = Bukkit.getOfflinePlayer(subscription.getPlayerId());
+    //If it is still overdue (they didnt pay it or failed to pay it)
+    if (subscription.isOverdue()) {
+      if (subscription.getDaysOverdue() > service.getDuration()) {
+        //They have not paid their subscription in over the duration
+        //Auto cancel it
+        try {
+          DatabaseHelper.getInstance().getDatabaseManager().getSubscriptionDao().delete(subscription);
+          Bukkit.getServer().dispatchCommand(Bukkit.getServer().getConsoleSender(), "mail send " +
+              player.getName() + " Your subscription to " + service.getName() + " has been cancelled since you didn't pay it in over " + service.getDuration()
+              + " days. This is because " + (subscription.isAutoPay() ? " you did not have enough money to automatically pay it." : " you did not manually renew it.")
```

Comment:
Shorten this message, getting way to close to max char limit. (Remember all users aren't idiots)

Reply:
Done.

Reply:
https://github.com/schlatt-co/stonks/blob/a34c87a8a16fdce21ba0462044aedb07a35af01b/src/main/java/dev/tycho/stonks/scheduledtasks/SubscriptionCheckTask.java#L101